### PR TITLE
Convert Slider to ES6 and resolve lint errors

### PR DESCRIPTION
**Pull request by adamrneary**
@ljharb this PR brings in the pieces from https://github.com/airbnb/rheostat/pull/35/commits/191744b9457029fdc80a9ee93fade394863be33b that survives post rebase. It eliminates 6 of the eslint exceptions and tees us up for the react-with-styles commit, which I will repost as a separate PR once this is in.

### CODE REVIEW

**Comment by adamrneary** [comment]
Note: This would be a patch semver, but i am not sure it warrants a release on its own.

**Comment by lencioni** [comment]
Looks like this maybe broke React 0.13 tests. https://travis-ci.org/airbnb/rheostat/jobs/168961319

File: src/Slider.jsx [hunk 1]
```diff
@@ -18,85 +18,91 @@ const has = Object.prototype.hasOwnProperty;
...
-  },
+function getHandleFor(ev) {
+  return Number(ev.currentTarget.getAttribute('data-handle-key'));
+}
 
-  getInitialState() {
-    const { max, min, values } = this.props;
+function killEvent(ev) {
+  ev.stopPropagation();
+  ev.preventDefault();
```
Comment (lencioni) [inline src/Slider.jsx]: Do we still need `cancelBubble` and `returnValue` settings in this function?

File: stories/ExampleSlider.jsx [hunk 2]
```diff
@@ -59,24 +59,22 @@ storiesOf('Slider', module)
...
-              zIndex: 3,
-            })}
-          />
-        );
-      }
+    function MyHandle({ style, ...passProps }) {
+      return (
+        <div
+          {...passProps}
+          style={Object.assign({}, style, {
```
Comment (lencioni) [inline stories/ExampleSlider.jsx]: You could use object spread here instead of Object.assign if you like.

**Comment by ljharb** [comment]
Do class-based components work in React 0.12? I know SFCs don't work in 0.13, so we'd need `prefer-stateless-function` to remain disabled.

**Comment by adamrneary** [comment]
@ljharb @lencioni good news: tests are now passing. bad news: i played around in codepen to confirm what we already knew: ES6 classes won't work with React 0.12. The original blog post for 0.13 touches on this https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html, and I was able to confirm by loading up the old 0.12 code.

So, do we think compatibility with 0.12 is a worthwhile constraint? And, when we get to the next commit, with react-with-styles have any issue with the legacy React.createClass() syntax?

If we think 0.12 is required and that react-with-styles won't have any issues, I can close this PR outright and just work on the latter…

**Comment by lencioni** [comment]
> So, do we think compatibility with 0.12 is a worthwhile constraint? 

I don't think it is worthwhile to continue support for React 0.12 in this project. 0.13 was released about a year and a half ago and folks who are still on that version can continue to use an older version of Rheostat. 

> And, when we get to the next commit, with react-with-styles have any issue with the legacy React.createClass() syntax?

I can't think of any reasons why react-with-styles won't work with React 0.12. I guess it is possible that something changed with context, but I'm not aware of anything like that.

**Comment by ljharb** [comment]
I don't mind dropping 0.12 - it would just be semver-major. I strongly feel we should support down to N - 2 - ie, continue to support 0.13 (which means no SFCs).

In general, the true statement "users can just stay on the older version!" doesn't hold up in practice. Any time we drop support, it hurts the migration path for those users. Hopefully nobody is still on 0.12 - and the current version of Rheostat supports 0.12 - 15 - so the migration path should be clear - but it's always worth considering.

**Comment by adamrneary** [comment]
Very good. If this is the case, the current state of this PR passes. We are testing on React 13, 14, and 15. It uses ES6 classes but not SFCs. Please confirm this is a reasonable landing place?

As for tagging this semver-major, I would agreed, but I would also suggest waiting until we complete the separate PR for react-with-styles before cutting that release.

**Comment by ljharb** [comment]
That sounds fine with me.

**Comment by adamrneary** [comment]
Polite ping. I think i got an informal thumbs up but need an official stamp.

**Comment by ljharb** [comment]
Should this PR thus also drop 0.12 support, in travis.yml and package.json?

I'd also prefer to cut a release before this is merged.

**Comment by adamrneary** [comment]
@ljharb done and done. I made that change in a separate commit and rebased to move that commit up front so that we have a more coherent path. let me know if I can adjust further — sorry for the churn!

**Comment by goatslacker** [comment]
LGTM

**Pull request merged**